File: GolangSolutions/Array/Q349_两个数组的交集.go
package Array

func intersection(nums1 []int, nums2 []int) []int {
	res := []int{}
	hashMap := make(map[int]int)
	for i := range nums1 {
		hashMap[nums1[i]]++
	}
	for i := range nums2 {
		if _, ok := hashMap[nums2[i]]; ok {
			res = append(res, nums2[i])
			delete(hashMap, nums2[i])
		}
	}

	return res
}

File: CPPSolutions/LeetCode/Array/Q349_两个数组的交集.cpp
#include <bits\stdc++.h>
#include <unordered_set>

using namespace std;

class Solution {
public:
    vector<int> intersection(vector<int> &nums1, vector<int> &nums2) {
        unordered_set<int> res, nums1_set(nums1.begin(), nums1.end());
        for (int i = 0; i < nums2.size(); i++)
            if (nums1_set.find(nums2[i]) != nums1_set.end())
                res.insert(nums2[i]);

        return vector<int>(res.begin(), res.end());
    }
};